dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePlugin.kt

```diff
@@ -184,17 +184,20 @@ class DdAndroidGradlePlugin : Plugin<Project> {
 
     internal fun resolveExtensionConfiguration(
         extension: DdExtension,
-        flavorName: String
+        variant: ApplicationVariant
     ): DdExtensionConfiguration {
-        val flavorConfig = extension.variants.findByName(flavorName)
-
-        return DdExtensionConfiguration().apply {
-            versionName = flavorConfig?.versionName ?: extension.versionName
-            serviceName = flavorConfig?.serviceName ?: extension.serviceName
-            site = flavorConfig?.site ?: extension.site
-            checkProjectDependencies = flavorConfig?.checkProjectDependencies
-                ?: extension.checkProjectDependencies
+        val configuration = DdExtensionConfiguration()
+        configuration.updateWith(extension)
+
+        val flavors = variant.productFlavors.map { it.name }
+        val iterator = VariantIterator(flavors)
+        iterator.forEach {
+            val config = extension.variants.findByName(it)
+            if (config != null) {
```

Comment:
I think it is useful to run also the opposite check - check entries defined in `extension.variants` which are missing in flavors/variants and show a warning if it is a case. this may happen because of typo, for example, and may go unnoticed for a very long time.